zarr/v3/metadata.py

```diff
@@ -154,20 +163,19 @@ class ShardingCodecIndexLocation(Enum):
 
 
 @frozen
-class CoreArrayMetadata:
-    shape: ChunkCoords
+class ChunkMetadata:
     chunk_shape: ChunkCoords
     data_type: DataType
     fill_value: Any
-    runtime_configuration: RuntimeConfiguration
+    runtime_configuration: RuntimeConfiguration = runtime_configuration("C")
```

Comment:
Following up on the comment above about the semantic target for this class, I'm not sure `runtime_configuration` is a good fit on a class that basically is an abstract (no data) description of an array. I'm pretty sure that`runtime_configuration` is included here because functions that consume this class need  a `runtime_configuration` argument, but I would argue that this is a problem for the callers of those functions, i.e. if code is calling a function that requires a `runtime_configuration`, then lets just make it part of the signature and callers can deal with it formally.

Reply:
that being said, we can deal with this kind of refactoring separately from this PR

Reply:
Fair enough, we can add `runtime_configuration` as separate argument to the codec signature.